hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieAppendHandle.java

```diff
@@ -326,12 +327,7 @@ private MetadataValues populateMetadataFields(HoodieRecord<T> hoodieRecord) {
   private void initNewStatus() {
     HoodieDeltaWriteStat prevStat = (HoodieDeltaWriteStat) this.writeStatus.getStat();
     // Make a new write status and copy basic fields over.
-    HoodieDeltaWriteStat stat = new HoodieDeltaWriteStat();
-    stat.setFileId(fileId);
-    stat.setPartitionPath(partitionPath);
-    stat.setPrevCommit(prevStat.getPrevCommit());
-    stat.setBaseFile(prevStat.getBaseFile());
-    stat.setLogFiles(new ArrayList<>(prevStat.getLogFiles()));
+    HoodieDeltaWriteStat stat = CommitUtils.getHoodieDeltaWriteStatFromPreviousStat(prevStat);
```

Comment:
How about we add a `copy` method to `HoodieDeltaWriteStat` ?

Reply:
Patiently waiting for @codope (as author of task) to answer this question.

Reply:
yes copy is more idiomatic.

Reply:
Ok, added a `copy` method to HoodieDeltaWriteStat